src/Sarif.UnitTests/Core/PropertyBagHolderTests.cs

```diff
@@ -19,6 +19,126 @@ internal class TestClass : PropertyBagHolder
     [TestClass]
     public class PropertyBagHolderTests
     {
+        private const string PropertyName = "prop";
```

Comment:
Added several more unit tests for the existing `Properties` APIs. Also includes a test for the new `RemoveProperty` API, which you will see below is used in the implementation of `Tags.Clear`.